### PR TITLE
NetworkPkg TcpDxe: Fixed system stuck on PXE boot flow in iPXE environment

### PR DESCRIPTION
**Pull request by ctjh881420**
Pkg-Module: NetworkPkg TcpDxe

This bug fix is based on the following commit "NetworkPkg TcpDxe: SECURITY PATCH"
REF: https://github.com/tianocore/edk2/commit/1904a64bcc18199738e5be183d28887ac5d837d7

Issue Description:
An "Invalid handle" error was detected during runtime when attempting to destroy a child instance of the hashing protocol. The problematic code segment was:

NetworkPkg\TcpDxe\TcpDriver.c
Status = Hash2ServiceBinding->DestroyChild(Hash2ServiceBinding, ​&mHash2ServiceHandle);

Root Cause Analysis:
The root cause of the error was the passing of an incorrect parameter type, a pointer to an EFI_HANDLE instead of an EFI_HANDLE itself, to the DestroyChild function. This mismatch resulted in the function receiving an invalid handle.

Implemented Solution:
To resolve this issue, the function call was corrected to pass mHash2ServiceHandle directly:

NetworkPkg\TcpDxe\TcpDriver.c
Status = Hash2ServiceBinding->DestroyChild(Hash2ServiceBinding, mHash2ServiceHandle);

This modification ensures the correct handle type is used, effectively rectifying the "Invalid handle" error.

Verification:
Testing has been conducted, confirming the efficacy of the fix. Additionally, the BIOS can boot into the OS in an iPXE environment.

Signed-off-by: Sam Tsai [Wiwynn] <sam_tsai@wiwynn.com>

### CODE REVIEW

**Comment by makubacki** [comment]
@ctjh881420, I see you added `Signed-off-by` to the PR description. Since commits are merged as-is to the `master` branch, please add more detail to the commit message itself (similar to your PR description) and add the `Signed-off-by` line there as well.

**Comment by ctjh881420** [comment]
@makubacki 
I have modified the commit message to include more detail, as suggested, and incorporated the Signed-off-by line directly within the commit message itself. Could you please review the revised commit message and confirm if it is correct? Thank you.

**Comment by lgao4** [comment]
@ctjh881420 , CI shows PatchCheck failure. Please resolve it.

**Comment by ctjh881420** [comment]
@lgao4 
CI shows PatchCheck failure as follows, but it was built an hour ago. I have just updated the commit message to include the Signed-off-by. Please rerun the CI build to confirm. Thank you.

Fix issue: System stuck on PXE boot flow in iPXE environment
The commit message format is not valid:
 * Missing Signed-off-by! (Note: this must be added by the code contributor!)

**Comment by bcran** [comment]
> @makubacki I have modified the commit message to include more detail, as suggested, and incorporated the Signed-off-by line directly within the commit message itself. Could you please review the revised commit message and confirm if it is correct? Thank you.

I'm not seeing any additional detail in the commit message. Have you force-pushed your change?

**Comment by ctjh881420** [comment]
@bcran @lgao4
Force-pushed my change to modify the commit message. Please review it and rerun the CI build. Thank you.

**Comment by makubacki** [comment]
@SaloniKasbekar, can you please approve the PR in GitHub if you have approved? As far as I can tell, this contribution was submitted after the PR transition and following the PR process so it requires approval in GitHub.

**Review by SaloniKasbekar — APPROVED** [review]

**Review by Flickdm — APPROVED** [review]

**Comment by makubacki** [comment]
Adding stewards as reviewers since there are no maintainers (only reviewers) for NetworkPkg.

**Review by mdkinney — APPROVED** [review]

**Comment by mdkinney** [comment]
@Mergifyio rebase

**Comment by ctjh881420** [comment]
https://github.com/Mergifyio rebase

**Pull request merged**
